api/host.py

```diff
@@ -147,7 +147,20 @@ def delete_host_list(
     tags=None,
     filter=None,
 ):
-    if not any([display_name, fqdn, hostname_or_id, insights_id, provider_id, provider_type, tags, filter]):
+    if not any(
+        [
+            display_name,
+            fqdn,
+            hostname_or_id,
+            insights_id,
+            provider_id,
+            provider_type,
+            registered_with,
+            staleness,
+            tags,
+            filter,
+        ]
+    ):
```

Comment:
Why did we abandon the `locals()` approach? I verified locally that this does the same thing as:
```suggestion
    if not any(locals()):
```

Reply:
There are more than one reasons.

1. `locals()` was to count the args passed in because at the time `if not any([...])` did not work due to my IDE or project being in bad state.
2.  The object returned by `locals()` structure is more complex than the simple dictionary of input arguments that deepcopy returns.  e.g. `len(locals())` would return the number of arguments where `if not locals()` would succeed for having all `None` In other I didn't feel confident in using it.
3.  The most important reason, `in not any([...])` is very explicit and easy to read.